tests/test_copies.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+import os
+import shutil
+from shallow_backup import _copy_file, copy_dir
+from constants import Constants
+
+TEST_DIR = 'test-directory'
+TEST_BACKUP_DIR = 'test-backup-dir'
+TEST_TEXT_FILE = 'test-file.txt'
+DIRS = [TEST_DIR, TEST_BACKUP_DIR]
+
+
+class TestCopyMethods:
+    """
+    Test the functionality of copying
+    """
+
+    def setup_method(self):
+        for directory in DIRS:
+            try:
+                os.mkdir(directory)
+            except FileExistsError:
+                shutil.rmtree(directory)
+                os.mkdir(directory)
+        f = open(TEST_TEXT_FILE, "w+")
+        f.close()
+
+    def teardown_method(self):
+        shutil.rmtree(TEST_DIR)
+        os.remove(TEST_TEXT_FILE)
+
+    def test_copy_file(self):
+        """
+        Test that copying a file is working as expected
+        """
+        process =_copy_file(TEST_TEXT_FILE, TEST_DIR)
+        assert process.returncode == 0
+        assert os.path.isfile(TEST_TEXT_FILE)
+        assert os.path.isfile(TEST_DIR + '/' + TEST_TEXT_FILE)
+
+    def test_copy_dir(self):
+        """
+        Test that copying a directory works as expected
+        """
+        dir_to_copy = '/test/'
+        os.mkdir(TEST_DIR + dir_to_copy)
+        process = copy_dir(TEST_DIR + dir_to_copy, TEST_BACKUP_DIR)
+        assert process.returncode == 0
+        assert os.path.isdir(TEST_DIR + dir_to_copy)
+        assert os.path.isdir(TEST_BACKUP_DIR + dir_to_copy)
+
+    @pytest.mark.parametrize('invalid', Constants.INVALIDS)
```

Comment:
I'm a little bit confused about this test. Two thoughts:

1. Does it make more sense to just access `Constants.INVALID_DIRS` inside the function instead of passing it in as a parameter?
2. Is this function passing a list into `copy_dir`? Shouldn't this be in the form of iterating through `INVALID_DIRS` and trying to copy each dir, and then checking the `process is None` condition for each?

Reply:
Great question. Originally when I was using unittest I was iterating through the list within the test. This is a great feature by pytest meaning that it will run a unique tests for every case in the list `Constants.INVALIDS`. This is better practices for a couple reasons but most importantly it creates a separate test for each unique case so that if one fails we know exactly what the problem is. We should aim to test the most atomic amount of functionality in our unit tests.

Reply:
Oh, cool! I didn't know that was a feature of `pytest`... 

I'm good with that rationale.